lib/resque/scheduler.rb

```diff
@@ -427,7 +427,15 @@ def logger
       private
 
       def enqueue_recurring(name, config)
-        if am_master
+        klass_name = config['class']
+        klass = Resque::Scheduler::Util.constantize(klass_name)
+        should_enqueue = Resque::Scheduler::Plugin.run_before_recurring_hooks(
+          klass
+        )
+
+        if am_master && !should_enqueue
+          log! "skipping queueing of #{config['class']} (#{name})"
+        elsif am_master && should_enqueue
```

Comment:
There is probably a simpler way to do this if. Maybe
`return if !am_master`
on the first line of the function, and they you only have to worry about `should_enque` here